src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1843,15 +1843,9 @@ private TypeDeclarationSyntax ParseMainTypeDeclaration(SyntaxList<AttributeListS
                         }
                     }
 
-                    if (openBrace.IsMissing)
-                    {
-                        closeBrace = SyntaxFactory.MissingToken(SyntaxKind.CloseBraceToken);
-                        closeBrace = WithAdditionalDiagnostics(closeBrace, this.GetExpectedTokenError(SyntaxKind.CloseBraceToken, this.CurrentToken.Kind));
-                    }
-                    else
-                    {
-                        closeBrace = this.EatToken(SyntaxKind.CloseBraceToken);
-                    }
+                    closeBrace = openBrace.IsMissing
+                        ? this.CreateMissingToken(SyntaxKind.CloseBraceToken, this.CurrentToken.Kind)
```

Comment:
note: CreateMissingToken already does WithAdditionalDiagnostics(... this.GetExpectedError(...))

so just replacing the duplcated code with the existing helper.